Automattic/wp-calypso
# Gutenboarding: Feature Picker

#### Changes proposed in this Pull Request

This PR will now display feature picker step. As user picks the features, the recommended plan on the plans page will change.

* Feature data:
  * Renamed to `onboarding-block/features/data.tsx`.
  * Feature list is now an object keyed by `featureId`.
  * Added `minSupportedPlan` property.
  * Name & descriptions are now translated.
* Onboarding
  * Added `selectedFeatures` reducer and `addFeature/removeFeature` actions to onboard store.
     * `selectedFeatures` is persistent (remembers when page refreshes).
  * Added `useRecommendedPlan()` and `useRecommendedPlanSlug()` hook.
  * Added `Step.Features` to `useStepNavigation()`.
* Plans grid:
  * `PlansGrid` component now accepts `recommendedPlan` param.
  * `PlansTable` component now accepts `recommendedPlanSlug` param.
  * Added recommendation hint ("Based on the features you selected") your on plans table.


#### Testing instructions

* Go through the `/new` flow.
* Pick different feature combinations:
  * It should recommend the correct plan based on your feature selections.
  * It should show "Based on the features you selected" hint.
  * It should show "Recommended For You" badge.
* Don't pick any features (click Skip for now):
  * It should recommend the default paid plan - Premium plan.
  * It should not show "Based on the features you selected" hint.
  * It should show "Popular" badge.
* Refresh the page. It should remember your feature selection.

#### Screenshot

![2020-07-24_17-19-06 (1)](https://user-images.githubusercontent.com/1287077/88407125-f1fd7500-cdd1-11ea-987c-329341dc3acf.gif)


Fixes part of #44428
